### DISCUSSION TITLE
Database connection doesn't close after command

### DISCUSSION THREAD

**Discussion by backspacerhino** [adonisjs/core#1407] [Help] [answered]
Given the next code for command:

```
...
@inject([null, null, 'Adonis/Lucid/Database'])
export default class TestCommand extends BaseCommand {
    public static commandName = 'test'
    public static description = 'test command'

    constructor(app: ApplicationContract, kernel: Kernel, private db: DatabaseContract) {
        super(app, kernel)
    }

    public static settings = {
        loadApp: true,
    }

    public async handle() {
        console.log('testing...');
        // The line below is what causes problems
        const users = await this.db.query().select('*').from('users');
        console.log('END');
    }
...
```

The command never finished if we have any Database connection opened.
If I remove that line then the command finishes

**Comment by thetutlage** [comment, marked as answer]
Expected. You have to close the database connection manually. AdonisJS cannot make the decision whether to forcefully close the connection or not. What if your command intentionally is a long running process?

```ts
await this.db.manager.closeAll()
```

**Reply by backspacerhino** [reply]
Thank you for the reply, now it closes properly. I tried looking in [Connection managments](https://preview.adonisjs.com/guides/database/connections-management) but couldn't find it. Could you please add this manual closing to the docs?

**Reply by thetutlage** [reply]
Send a PR please :)
